pkg/catalog/manager/manager.go

```diff
@@ -243,24 +241,38 @@ func (m *Manager) ValidateKubeVersion(template *v3.CatalogTemplateVersion, clust
 }
 
 func (m *Manager) ValidateRancherVersion(template *v3.CatalogTemplateVersion) error {
-	rancherMin := template.Spec.RancherMinVersion
-	rancherMax := template.Spec.RancherMaxVersion
-
 	serverVersion := settings.ServerVersion.Get()
 
 	// don't compare if we are running as dev or in the build env
 	if !utils.ReleaseServerVersion(serverVersion) {
 		return nil
 	}
 
-	if rancherMin != "" && !mVersion.Compare(serverVersion, rancherMin, ">=") {
-		return fmt.Errorf("rancher min version not met")
-	}
+	serverVersion = strings.TrimPrefix(serverVersion, "v")
 
-	if rancherMax != "" && !mVersion.Compare(serverVersion, rancherMax, "<=") {
-		return fmt.Errorf("rancher max version exceeded")
+	versionRange := ""
+	if template.Spec.RancherMinVersion != "" {
+		versionRange += " >=" + strings.TrimPrefix(template.Spec.RancherMinVersion, "v")
+	}
+	if template.Spec.RancherMaxVersion != "" {
+		versionRange += " <=" + strings.TrimPrefix(template.Spec.RancherMaxVersion, "v")
```

Comment:
Some use `x.y.z-alpha1` and `x.y.99-0` together, like the following:
https://github.com/rancher/system-charts/blob/5a757daa2a82ab4ffeedadbf1ad68d9ec1661524/charts/rancher-logging/0.2.5/questions.yml#L1-L2
whereas others use `x.y.z-rc1` and `x.y.z`, like so:
https://github.com/rancher/system-charts/blob/9d3c12a9704b5501baa87193a07fb9f953c465af/charts/rancher-monitoring/v0.2.1/questions.yml#L1-L2
I think regardless it's safe to say we need more unit and integration tests covering this, because the integration tests at https://github.com/rancher/integration-test-charts only use `x.y.z` for both min and max version. This is also vastly different from the actual catalog charts, as they all use `x.y.z-(alpha/rc)1` (except for one [dynatrace chart](https://github.com/rancher/helm3-charts/blob/136a0a384c31a88393bdc56770ba8c848aa2eac9/charts/dynatrace-operator/v0.2.1/questions.yml#L4)).

Also to answer you're question, yes, blang has no such restriction on prereleases, as the unit tests (not the integration tests) would fail. This is achievable with https://github.com/Masterminds/semver, but would require adding prereleases to the max and min version that do not have it. 

A real concrete example is the following:
```
$ vert ">=2.5.6-rc1,<=2.5.7" 2.5.6-rc1
$ echo $?
1
```
Which comes from the monitoring chart listed above. This works as is right now with blang/semver, which I can only assume is what we want otherwise most of the catalogs would disappear for an rc, but it also works with go-version, so I guess the question is are we going to update all of the charts with min and max version to add `-0` (either in the charts or in the validation check) if there is no prerelease for the given min/max?

Reply:
@jakefhyde , thank you for detailed explanation and research on what ranges our charts use.
I've played with previous github.com/mcuadros/go-version based code and new github.com/blang/semver based one and found one mismatch, but I think it's actually correct.
Basically in go-version 2.5.8 or 2.5.8-rc1 fits 2.5.7-alpha1 to 2.5.8-0 range while it doesn't in blang's case - https://play.golang.org/p/hPo4E3ZEbnL. The latter is more correct as the intention of <=2.5.8-0 is to essentially make the upper range not include anything from 2.5.8 release line.

Bottom line I think we're fine as far as the actual code, but it would be nice to update at least unit tests to use realistic ranges we're using rather than ambiguous ranges as we use here https://github.com/jakefhyde/rancher/blob/833a37a62556c448c3599573b107696ea2033e52/pkg/catalog/manager/manager_test.go#L15-L32 as for example 2.1.0 fits 2 ranges.

I know it's a bit beyond what the issue is immediately about, but we want to do the right thing here and have unit tests represent the realistic ranges we use.